statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,721 @@
+
+        if hess_arr_main is None or hess_arr_infl is None:
+            return approx_hess(params, self.loglike)
+
+        dim = self.k_exog + self.k_inflate
+
+        hess_arr = np.zeros((dim, dim))
+
+        hess_arr[:self.k_inflate,:] = hess_arr_infl
+        hess_arr[self.k_inflate:,self.k_inflate:] = hess_arr_main
+
+        tri_idx = np.triu_indices(self.k_exog + self.k_inflate, k=1)
+        hess_arr[tri_idx] = hess_arr.T[tri_idx]
+
+        return hess_arr
+
+    def predict(self, params, exog=None, exog_infl=None, exposure=None,
+                offset=None, which='mean'):
+        """
+        Predict response variable of a count model given exogenous variables.
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+        exog : array, optional
+            A reference to the exogenous design.
+            If not assigned, will be used exog from fitting.
+        exog_infl : array, optional
+            A reference to the zero-inflated exogenous design.
+            If not assigned, will be used exog from fitting.
+        offset : array, optional
+            Offset is added to the linear prediction with coefficient equal to 1.
+        exposure : array, optional
+            Log(exposure) is added to the linear prediction with coefficient
+            equal to 1. If exposure is specified, then it will be logged by the method.
+            The user does not need to log it first.
+        which : string, optional
+            Define values that will be predicted.
+            'mean', 'poisson-mean', 'linear', 'mean-nonzero', 'prob-zero, 'prob'
+            Default is 'mean'.
+
+        Notes
+        -----
+        """
+        if exog is None:
+            exog = self.exog
+            offset = getattr(self, 'offset', 0)
+            exposure = getattr(self, 'exposure', 0)
+
+        if exog_infl is None:
+            exog_infl = self.exog_infl
+
+        if exposure is None:
+            exposure = 0
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = 0
+
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        if self.infl == 'logit':
+            prob_poisson = 1 / (1 + np.exp(np.dot(exog_infl, params_infl)))
+        elif self.infl == 'probit':
+            raise NotImplemented('Predict for Probit inflation not implemented')
+
+        lin_pred = np.dot(exog, params_main[:self.exog.shape[1]]) + exposure + offset
+        
+        tmp_exog = self.model_main.exog
+        tmp_endog = self.model_main.endog
+        self.model_main.exog = exog
+        self.model_main.endog = np.zeros((exog.shape[0]))
+        llf = self.model_main.loglikeobs(params_main)
+        self.model_main.exog = tmp_exog
+        self.model_main.endog = tmp_endog
+
+        prob_zero = (1 - prob_poisson) + prob_poisson * np.exp(llf)
```

Comment:
np.exp(llf)  looks only correct for Poisson and not for the others.
it should be pmf(0) of main model
(insufficient test coverage?

Reply:
See 7 previous line, llf in this case will be the same as pmf.
I tried to use `self.distribution.pmf` but in this case i'll have to rewrite predict function for each model because of different amount of parameters.